### PR TITLE
Added deployment annotations to chart

### PR DESCRIPTION
**Pull request by timetinytim**
Added the ability to define custom annotations for deployment objects.

### CODE REVIEW

**Review by renchap — CHANGES_REQUESTED** [review]
You may want to allow deployment-specific annotations as well, as we currently do for pod annotations (so its consistent).

For example for web:

```
{{- with (default .Values.deploymentAnnotations .Values.mastodon.web.deploymentAnnotations) }}
```

File: values.yaml [hunk 1]
```diff
@@ -408,6 +408,10 @@ serviceAccount:
   # If not set and create is true, a name is generated using the fullname template
   name: ""
 
+# Custom annotations to apply to all created deployment objects. These can be
+# used to help mastodon interact with other services in the cluster.
+deployAnnotations: {}
```
[the remark this answers is not in the document]
Comment (timetinytim) [reply values.yaml]: Ah both are good calls. Updated!

**Comment by timetinytim** [comment]
@renchap Apologies for bothering, but if everything looks good now, I'd like to get this merged so I can ensure everything stays up to date

**Review by renchap — APPROVED** [review]
Looks good! 👍

**Pull request merged**